### DISCUSSION TITLE
Are there already any network and persistence implementations available?

### DISCUSSION THREAD

**Discussion by rozek** [streamich/json-joy#571] [Q&A]
First of all: thank you very much for your impressive work!

By looking through JSON-Joy, however, I haven't found any network and persistence implementations for your library.

Am I blind? Am I missing something? Or did I misunderstood JSON-Joy completely?

Thanks in advance for your answer!

With greetings from Germany,

Andreas Rozek

**Comment by streamich** [comment]
Persistence and networking are work in progress. Right now you would need to manage those yourself. However, going forward there will be ready made solution, for browser-server communication [Reactive RPC](https://jsonjoy.com/specs/json-rx) will be used, for persistence there will an off-the-shelf implementation for storage in browser.

**Comment by streamich** [comment]
@rozek There different networking (HTTP, websocket, peer-to-peer, local-first) and storage (Noder, browser, IndexedDB, files) protocols. Which ones are you interested in?

**Reply by rozek** [reply]
WebSocket networking and IndexedDB persistence would be fine for the beginning.

Or some basic code which I could use to implement those as I already wrote some (storage) providers for Yjs (e.g., [y-localforage](https://github.com/rozek/y-localforage)). I am, however, a complete JSON-Joy beginner!

**Reply by rozek** [reply]
BTW: I know issue [#503](https://github.com/streamich/json-joy/issues/503) which contains some basic information about how one could implement a networking protocol between a client and a websocket server

**Reply by streamich** [reply]
I have implemented a demo server, can be found [here](https://github.com/jsonjoy-com/reactive-rpc/tree/master/src/__demos__/json-crdt-server); and currently working on the browse client which persists locally into OPFS.

**Reply by rozek** [reply]
Thank you very much for your effort!

Since it will probably take some time to analyze and understand the code (and "Reactive-RPC" - and the "MemoryStore"), I'll have a look at it this coming weekend.
